docs/tableconfig_schema.rst

```diff
@@ -0,0 +1,204 @@
+..
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+..
+..   http://www.apache.org/licenses/LICENSE-2.0
+..
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+..
+
+Table Config
+============
+
+Table Config
+-------------
+
+Introduction
+~~~~~~~~~~~~
+
+Using tables is how Pinot serves and organizes data. There are many settings in the table config which will influence how Pinot operates. The first and most significant distinction is using an offline versus a realtime table.
+
+An offline table in Pinot is used to host data which might be periodically uploaded - daily, weekly, etc. A realtime table, however, is used to consume data from incoming data streams and serve this data in a near-realtime manner. This might also be referred to as nearline or just plain 'realtime'.
+
+In this section a sample table configuration will be shown and all sections will be explained and if applicable have appropriate sections linked to for further explanation of those corresponding Pinot features.
+
+Sample table config and descriptions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A sample table config is shown below which has sub-sections collasped. The sub sections will be described individually in following sections.
```

Comment:
It will be good to call out the mandatory sections and opitonal ones. For ex, tableName, tableType and segmentConfig are mandatory.

Reply:
Are these the only mandatory sections? I don't know what is mandatory or not in this case since generally use all of them.

Reply:
Sunitha has pointed me to this PR by @Jackie-Jiang 
https://github.com/apache/incubator-pinot/pull/3993

I am waiting for a response from Jackie to help me parse what exactly is required.

Reply:
xajiang [12:48 PM]
You can check `TableConfig.validate()` method

  ```public void validate() {
    Preconditions.checkState(_tableName != null, "Table name is missing");
    Preconditions.checkState(_tableType != null, "Table type is missing");
    Preconditions.checkState(_validationConfig != null, "Validation config is missing");
    Preconditions.checkState(_tenantConfig != null, "Tenant config is missing");
    Preconditions.checkState(_indexingConfig != null, "Indexing config is missing");
    Preconditions.checkState(_customConfig != null, "Custom config is missing");
  }```

xajiang [12:51 PM]
  ```public static final String TABLE_NAME_KEY = "tableName";
  public static final String TABLE_TYPE_KEY = "tableType";
  public static final String VALIDATION_CONFIG_KEY = "segmentsConfig";
  public static final String TENANT_CONFIG_KEY = "tenants";
  public static final String INDEXING_CONFIG_KEY = "tableIndexConfig";
  public static final String CUSTOM_CONFIG_KEY = "metadata";```

Reply:
@sunithabeeram Let me know if this commit fulfills what you are asking for - 

https://github.com/apache/incubator-pinot/pull/3975/commits/599bc90a29812e6ddeef7fca3f574519287b3ec5?short_path=192dc63#diff-192dc63b00c8f5f79817d3a1f931fdfd